### PR TITLE
feat(VIcon): Add 'tag' prop to VIcon that sets a custom tag to be used

### PR DESCRIPTION
**Pull request by FRFlor**
## Description
Add a new optional prop to VIcon, the 'tag' prop, that specifies a custom tag to use in the
component, similar to the tag prop from VContainer. 

For the sake of backwards compatibility with existing Vuetify projects, this PR keeps the standard behavior of using the `<i>` tag if none is provided. However, it also gives the option for the developer to pass a new tag prop. If this prop is provided, the V-icon component will use the element specified as the wrapper. 

## Motivation and Context
The usage of `<i>`  is not recommended under the guidelines of accessibility (WCAG A/AA), since the`<i>` tag is not handled well by some screenreaders.

Even though these `<i>` tags are being used for icons on the VIcon component, this still leads to warnings on Accessibility tests.

Using the Accessibility Checker website, we can see multiple instances of errors from the icons used on the page. 

https://achecker.ca/ . 

On either https://vuetifyjs.com/en/ or  https://www.felipeflor.com will show that every instance of v-icon being used triggers a warning. 

image

## How Has This Been Tested?
The changes have been implemented with a TDD approach. Two new unit tests have been added with this new commit, one that assures the component will remain using the `<i>` tag if none is provided, and another test that assures the component will use any given tag provided by the developer. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by FRFlor** [comment]
@sh7dm Rebased with next

**Comment by sh7dm** [comment]
@FRFlor thanks. Have you tested it with different icon sets such as Material Icons, Material Design Icons, Font Awesome 4/5 and SVG?

File: packages/vuetify/src/components/VIcon/VIcon.ts [hunk 1]
```diff
@@ -34,7 +34,12 @@ const VIcon = mixins(
     disabled: Boolean,
     left: Boolean,
     right: Boolean,
-    dense: Boolean
+    dense: Boolean,
+    tag: {
+      type: String,
+      required: false,
+      default: 'i'
+    },
```
Comment (sh7dm) [inline packages/vuetify/src/components/VIcon/VIcon.ts]: @FRFlor please remove trailing comma

Comment (FRFlor) [reply packages/vuetify/src/components/VIcon/VIcon.ts]: Sorry about that, the issue should be fixed now.

**Comment by FRFlor** [comment]
> @FRFlor thanks. Have you tested it with different icon sets such as Material Icons, Material Design Icons, Font Awesome 4/5 and SVG?

I'll do a more thorough test and will let you know.

**Comment by FRFlor** [comment]
@sh7dm 
I did more tests on a personal project and the component seems to be working fine regardless of the source of the icons.  

Everything seems okay. 

Thanks for all the help, really appreciated it!

**Pull request merged**